platform/fabric/core/generic/delivery.go

```diff
@@ -8,69 +8,145 @@ package generic
+
+	deliveryService *delivery2.Delivery
+}
+
+func NewDeliveryService(
+	channel string,
+	channelConfig driver.ChannelConfig,
+	hasher hash.Hasher,
+	networkName string,
+	localMembership driver.LocalMembership,
+	configService driver.ConfigService,
+	peerManager delivery2.PeerManager,
+	ledger driver.Ledger,
+	waitForEventTimeout time.Duration,
+	txIDStore driver.TXIDStore,
+	callback delivery2.Callback,
+) (*DeliveryService, error) {
+	deliveryService, err := delivery2.New(
+		networkName,
+		channelConfig,
+		hasher,
+		localMembership,
+		configService,
+		peerManager,
+		ledger,
+		callback,
+		txIDStore,
+		channelConfig.CommitterWaitForEventTimeout(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &DeliveryService{
+		channel:             channel,
+		channelConfig:       channelConfig,
+		hasher:              hasher,
+		NetworkName:         networkName,
+		LocalMembership:     localMembership,
+		ConfigService:       configService,
+		PeerManager:         peerManager,
+		Ledger:              ledger,
+		waitForEventTimeout: waitForEventTimeout,
+		deliveryService:     deliveryService,
+	}, nil
+}
+
+func (c *DeliveryService) Start(ctx context.Context) error {
+	c.deliveryService.Start(ctx)
 	return nil
 }
 
-func (c *Channel) Scan(ctx context.Context, txID string, callback driver.DeliveryCallback) error {
+func (c *DeliveryService) Stop() {
+	c.deliveryService.Stop()
+}
+
+func (c *DeliveryService) Scan(ctx context.Context, txID string, callback driver.DeliveryCallback) error {
 	vault := &fakeVault{txID: txID}
-	deliveryService, err := delivery2.New(c.ChannelConfig, c.SP, c.Network, func(block *common.Block) (bool, error) {
-		for i, tx := range block.Data.Data {
-			validationCode := ValidationFlags(block.Metadata.Metadata[common.BlockMetadataIndex_TRANSACTIONS_FILTER])[i]
+	deliveryService, err := delivery2.New(
+		c.NetworkName,
+		c.channelConfig,
+		c.hasher,
+		c.LocalMembership,
+		c.ConfigService,
+		c.PeerManager,
+		c.Ledger,
+		func(block *common.Block) (bool, error) {
+			for i, tx := range block.Data.Data {
+				validationCode := ValidationFlags(block.Metadata.Metadata[common.BlockMetadataIndex_TRANSACTIONS_FILTER])[i]
 
-			if pb.TxValidationCode(validationCode) != pb.TxValidationCode_VALID {
-				continue
-			}
+				if pb.TxValidationCode(validationCode) != pb.TxValidationCode_VALID {
+					continue
+				}
 
-			env, err := protoutil.UnmarshalEnvelope(tx)
-			if err != nil {
-				logger.Errorf("Error getting tx from block: %s", err)
-				return false, err
-			}
-			payload, err := protoutil.UnmarshalPayload(env.Payload)
-			if err != nil {
-				logger.Errorf("[%s] unmarshal payload failed: %s", c.ChannelName, err)
-				return false, err
-			}
-			channelHeader, err := protoutil.UnmarshalChannelHeader(payload.Header.ChannelHeader)
-			if err != nil {
-				logger.Errorf("[%s] unmarshal Channel header failed: %s", c.ChannelName, err)
-				return false, err
-			}
+				env, err := protoutil.UnmarshalEnvelope(tx)
```

Comment:
The same logic is also under fabric/core/generic/committer/committer.go. We could extract and re-use.